sdk/formrecognizer/Azure.AI.FormRecognizer/samples/Sample_AnalyzeDocument.md

```diff
@@ -0,0 +1,204 @@
+# Analyze a general document
+
+This sample demonstrates how to extract text, tables, styles, selection marks like radio buttons, entities, key-value pairs, and layout information from general documents, without the need to train a model.
+
+To get started you'll need a Cognitive Services resource or a Form Recognizer resource.  See [README][README] for prerequisites and instructions.
+
+## Creating a `DocumentAnalysisClient`
+
+To create a new `DocumentAnalysisClient` you need the endpoint and credentials from your resource. In the sample below you'll use a Form Recognizer API key credential by creating an `AzureKeyCredential` object, that if needed, will allow you to update the API key without creating a new client.
+
+You can set `endpoint` and `apiKey` based on an environment variable, a configuration setting, or any way that works for your application.
+
+```C# Snippet:CreateDocumentAnalysisClient
+string endpoint = "<endpoint>";
+string apiKey = "<apiKey>";
+var credential = new AzureKeyCredential(apiKey);
+var client = new DocumentAnalysisClient(new Uri(endpoint), credential);
+```
+
+## Analyze a general document from a URI
+
+To analyze a given file at a URI, use the `StartAnalyzeDocumentFromUri` method and pass `prebuilt-document` as the model ID. The returned value is an `AnalyzeResult` object containing data about the submitted document.
+
+```C# Snippet:FormRecognizerAnalyzeDocumentFromUriAsync
+string fileUri = "<fileUri>";
+
+AnalyzeDocumentOperation operation = await client.StartAnalyzeDocumentFromUriAsync("prebuilt-document", fileUri);
+
+await operation.WaitForCompletionAsync();
```

Comment:
In V2 we used operation extensions to be able to do:
```cs
var operation = await client.StartAnalyzeDocumentAsync("<...>", stream).WaitForCompletionAsync();
```
in one step.

Do we still see the need in supporting this? I think it would be a discussion for beta 2, but it could make sense to add it if we think we have customers that are used to the shorter pattern.

Reply:
It's a nice feature that I frequently wish we could have in JS. I've thought about extending the `Promise` class in JavaScript to support something like `const result = await client.beginAnalyzeDocument(...).pollUntilDone();`. Was that feature in the previous package implemented as an extension method of `Task<AnalyzeDocumentOperation>` (or equivalent)?

Reply:
From what I've seen, this is the only library that has that pattern in all .NET
For beta.1 def leave as is, but follow up with architects about it.

Reply:
> Was that feature in the previous package implemented as an extension method of `Task<AnalyzeDocumentOperation>` (or equivalent)?

Correct, [here's the implementation we used for V2](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/formrecognizer/Azure.AI.FormRecognizer/src/OperationExtensions.cs). As Mariana said, I never saw other libraries using it.

> For beta.1 def leave as is, but follow up with architects about it.

Created https://github.com/Azure/azure-sdk-for-net/issues/24346 for tracking.